consensus/src/dag/dag_state_sync.rs

```diff
@@ -256,7 +256,10 @@ impl DagStateSynchronizer {
 
         self.execution_client.sync_to(commit_li).await?;
 
-        Ok(Arc::into_inner(sync_dag_store).unwrap())
+        match Arc::try_unwrap(sync_dag_store) {
+            Ok(inner) => Ok(inner),
+            Err(_) => bail!("Failed to unwrap Arc, more than one strong reference exists"),
```

Comment:
this is also invariant violation, we probably should fail fast instead of erroring out.

Reply:
Do you mean to panic when you say "fail quickly", right?

Reply:
yes